modules/lang-expression/src/main/java/org/elasticsearch/script/expression/ExpressionExecutableScript.java

```diff
@@ -55,8 +55,8 @@ public ExpressionExecutableScript(Expression expression, Map<String, Object> var
             functionValuesMap.put(variableName, functionValuesArray[functionValuesIndex]);
         }
 
-        for (String varsName : vars.keySet()) {
-            setNextVar(varsName, vars.get(varsName));
+        for (Map.Entry<String, Object> entry : vars.entrySet()) {
+            setNextVar(entry.getKey(), vars.get(entry.getKey()));
```

Comment:
This doesn't seem to take advantage of the entry value?

Reply:
Yes, you are right. I fixed it.